code/modules/admin/battle_royale.dm

```diff
@@ -0,0 +1,429 @@
+		CHECK_TICK
+		var/mob/living/carbon/human/H = new(pod)
+		ADD_TRAIT(H, TRAIT_PACIFISM, BATTLE_ROYALE_TRAIT)
+		//Assistant gang
+		H.equipOutfit(/datum/outfit/job/assistant)
+		//Give them a spell
+		H.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/knock)
+		H.key = key
+		//Give weapons key
+		var/obj/item/implant/weapons_auth/W = new
+		W.implant(H)
+		players += H
+		to_chat(M, "<span class='notice'>You have been given knock and pacafism for 30 seconds.</span>")
+	new /obj/effect/DPtarget(spawn_turf, pod)
+	SEND_SOUND(world, sound('sound/misc/airraid.ogg'))
+	to_chat(world, "<span class='boldannounce'>A 30 second grace period has been established. Good luck.</span>")
+	to_chat(world, "<span class='boldannounce'>WARNING: YOU WILL BE GIBBED IF YOU LEAVE THE STATION Z-LEVEL!</span>")
+	to_chat(world, "<span class='boldannounce'>[players.len] people remain...</span>")
+	//Start processing our world events
+	START_PROCESSING(SSprocessing, src)
+	addtimer(CALLBACK(src, .proc/end_grace), 300)
+	generate_basic_loot(100)
+
+/datum/battle_royale_controller/proc/end_grace()
+	for(var/mob/M in GLOB.player_list)
+		M.RemoveSpell(/obj/effect/proc_holder/spell/aoe_turf/knock)
+		REMOVE_TRAIT(M, TRAIT_PACIFISM, BATTLE_ROYALE_TRAIT)
+		to_chat(M, "<span class='greenannounce'>You are no longer a pacafist. Be the last [M.gender == MALE ? "man" : "woman"] standing.</span>")
+
+//==================================
+// EVENTS / DROPS
+//==================================
+
+/datum/battle_royale_controller/proc/generate_basic_loot(amount=1)
+	for(var/i in 1 to amount)
+		send_item(pick(GLOB.battle_royale_basic_loot))
+		stoplag()
+
+/datum/battle_royale_controller/proc/generate_good_drop()
+	var/list/good_drops = list()
+	for(var/i in 1 to rand(1,3))
+		good_drops += pick(GLOB.battle_royale_good_loot)
+	send_item(good_drops, announce = "Incomming extended supply materials.", force_time = 600)
+
+/datum/battle_royale_controller/proc/generate_endgame_drop()
+	var/obj/item = pick(GLOB.battle_royale_insane_loot)
+	send_item(item, announce = "We found a weird looking package in the back of our warehouse. We have no idea what is in it, but it is marked as incredibily dangerous and could be a superweapon.", force_time = 1800)
+
+/datum/battle_royale_controller/proc/send_item(item_path, style = STYLE_BOX, announce=FALSE, force_time = 0)
+	if(!item_path)
+		return
+	var/turf/target = get_safe_random_station_turf()
+	var/obj/structure/closet/supplypod/battleroyale/pod = new()
+	if(islist(item_path))
+		for(var/thing in item_path)
+			new thing(pod)
+	else
+		new item_path(pod)
+	if(force_time)
+		pod.fallDuration = force_time
+	new /obj/effect/DPtarget(target, pod)
+	if(announce)
+		priority_announce("[announce] \nExpected Drop Location: [get_area(target)]\n ETA: [force_time/10] Seconds.", "High Command Supply Control")
+
+//==================================
+// WORLD BORDER
+//==================================
+
+/obj/effect/death_wall
```

Comment:
What if someone teleports outside?

Reply:
That's handled in the process() of the controller

Reply:
Areas outside the death wall should probably have their area replaced with one that calls death() on enter.

Reply:
That seems a lot more expensive that just checking if the mobs x is lower than a certain value since when the wall is at max size it needs to update 1000 areas at once. The only other viable solution would be using a component that checks when the mob moves however, simple less than checks are pretty cheap and guaranteed to never fail.

Reply:
Why instadeath? Fortnite storm damage is more similar to brute damage over time, perhaps soft crit within 15 seconds. This allows making the tactical decision to tradeoff taking some small storm damage for a few seconds to obtain better positioning/grab an item/etc.

Reply:
@Victor239 Because it'll be funny watching people get gibbed when they think they'll be safe for a few moments.

Reply:
And you can aggressive grab throw people into it to have them violently explode